Empty content pages in Reader: avoid stuck secondary button

jeherve opened the pull request:
Fixes CML-916

## Proposed Changes

Handle margin a bit better on pages with empty content and a secondary button.

**Before**

<img width="753" height="275" alt="Screenshot 2025-09-23 at 18 13 26" src="https://github.com/user-attachments/assets/f616caac-3e9d-422d-8f19-abf66dfce10b" />

**After**

<img width="748" height="255" alt="Screenshot 2025-09-23 at 18 10 11" src="https://github.com/user-attachments/assets/b5ba8d76-19ea-4c2a-a645-47ffcd151324" />
<img width="661" height="255" alt="Screenshot 2025-09-23 at 18 10 45" src="https://github.com/user-attachments/assets/f96bad60-83d6-4af4-9418-f308900ddbed" />

## Testing Instructions

* Go to https://wordpress.com/tag/music-asdf
    * Notice the issue
* Go to http://calypso.localhost:3000/tag/music-asdf
    * Notice that the buttons are now separated

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

jeherve committed:
Empty content pages in Reader: avoid stuck secondary button

Fixes CML-916

Addison-Stavlo commented on client/components/empty-content/style.scss:
```diff
@@ -47,11 +47,19 @@
 
 .empty-content__action {
-	margin: 0 0 10px 10px;
+	margin: 0 0 10px 0;
```
It looks like this wasn't working because this style rule has the same specificity as `.button` general style, and is overwritten in the cascade loading order. 

We should probably change the selector to add 1 more level of specificity, so other styles are not overwritten unexpectedly as well. That would probably allow us to avoid the extra first-child/last-child logic.

Aside - your current styles seem to render a spacing similar to if this margin were also changed to `0 10px 10px` (i.e. short for `0 10px 10px 10px`),  which I think also does look better than when that original margin rule is actually active.

jeherve replied:
That's a good point, I should have taken a much simpler approach. I just pushed 19fa36d020161d78c7103e32ee9bc6f4013f3179 to address that.

> your current styles seem to render a spacing similar to if this margin were also changed to 0 10px 10px (i.e. short for 0 10px 10px 10px), which I think also does look better than when that original margin rule is actually active.

Since I reverted my changes we're back at `0 0 10px 10px` ; I figured I would leave it like that now.

Addison-Stavlo reviewed (commented):
It works as expected!  

A comment though, it looks like the original margin style was being overwritten due to lack of specificity. If we get the specificity correct, we could probably just control this with the margin property and avoid the extra first/last child business?

jeherve committed:
Simplify by increasing specificity

See https://github.com/Automattic/wp-calypso/pull/105982/files#r2375785535

jeherve committed:
Merge remote-tracking branch 'origin/trunk' into fix/empty-content-secondary-stuck

Addison-Stavlo commented on client/components/empty-content/index.tsx:
```diff
@@ -41,7 +41,7 @@ export default function EmptyContent( props: EmptyContentProps ): JSX.Element {
 				<Button
 					primary
-					className="empty-content__action"
+					className="empty-content__action button"
```
Did we need to add this extra `button` class for it to work? I thought that Button component was already adding that class in the output, which was the reason for the style conflict at the same specificity level. We could probably avoid adding extra classes to these.

jeherve replied:
Ah yes you're right, I should have checked the component. I added the class based on other usage, but that's not really needed. I removed that in de1055939bfd1e6e8fc1ffda60a145577246f31e

Addison-Stavlo replied:
> I removed that in https://github.com/Automattic/wp-calypso/commit/de1055939bfd1e6e8fc1ffda60a145577246f31e

Oh my how interesting. It looks like it already existed on one button implementation in this component, and not the other. And you just removed it from the one that existed before and not the one we added. 🙃 How weird it was setup like that, but its not like it makes a functional difference. 🤔   Anyways, feel free to update that as you see fit.

jeherve replied:
Ah I meant to remove it here too. Done in 2647b096787e88229181eb06d12636c9b75defe8
, and I'll merge this.

Addison-Stavlo commented on client/components/empty-content/style.scss:
```diff
@@ -46,7 +46,7 @@
 }
 
-.empty-content__action {
+.button.empty-content__action {
```
🤔  I wonder if we should do something like:

```
.empty-content__action,
button.empty-content__action {
```
(thats with `button` being the element selector and not the class selector).

After looking deeper, it seems like there are a small handful of ad-hoc places that defined styles for empty content action. Some maybe for their own reasons and style, others probably because these styles were being overwritten and they needed to get their specific implementation to look right.  

Some of these may also be on the same level of specificity as `.button.empty-content__action` (e.g. [this one](https://github.com/Automattic/wp-calypso/blob/trunk/client/login/magic-login/style.scss#L161)), which could lead to inconsistent styles based on css loading order if we also use the dual class specificity here. 😵 

If we do the above suggestion, it would probably mean:
* we don't need to also add the `button` class to other elements in this PR.
* the `.empty-content__action` selector could cover odd cases any cases like anchors using that class without the button class also.
* the `button.empty-content__action` would be stronger than the specificity of the generic `.button` styles supplied by the a8c button component's css rules.
* the `button.empty-content__action` would not interfere with other custom implementations that use 2 class names or more.

Perhaps we could remove a lot of the style rules that these other places are re-defining. Some of them are slightly different, I wonder if that is really intended or if we can can remove them to be consistent from this components definition. That may be more than you want to look into in this PR though.

Addison-Stavlo replied:
> any cases like anchors using that class without the button class also.

Oh no, I forgot the reverse concern. There are anchors that use the button class along with this. 🙃  So I guess that idea would then be something like:

```
.empty-content__action,
button.empty-content__action,
a.empty-content__action {
```
for them as well? 😵 🤔 

Its all in a bit of a fun and mildly confusing state it looks like. 🙃

jeherve replied:
That should cover everything indeed! Pushed that in de1055939bfd1e6e8fc1ffda60a145577246f31e

jeherve committed:
Update how we target buttons

- No need to explicitely provide the button class, it's added by the component.
- Let's ensure we cover all uses of the class.

Co-authored-by: Addison Stavlo <Stavz01@gmail.com>

Addison-Stavlo reviewed (approved):
Looks and works well.

minor comment around the classname thing to do with what you will! 😆 🙃

jeherve committed:
Remove additional unnecessary class
